src/articles/dto/articles-response.dto.ts

```diff
@@ -0,0 +1,41 @@
+import { Article } from '@prisma/client';
+import { Expose, Transform, TransformFnParams } from 'class-transformer';
+import { IsNotEmpty, IsOptional, IsString } from 'class-validator';
+
+function shortenText({ value: text }: TransformFnParams) {
+  if (!text) {
+    return null;
+  }
+  if (text.length > 100) {
+    return `${text.substring(0, 98)}${'.'.repeat(3)}`;
+  }
+  return text;
+}
+
+export function capitalizeTitle({ value }: TransformFnParams) {
+  return value[0].toUpperCase() + value.substring(1);
+}
+
+function getTextLength({ obj }: TransformFnParams) {
+  if (!obj.text) {
+    return null;
+  }
+  return obj.text.length;
+}
+
+export class ArticlesResponseDto implements Article {
```

Comment:
The validation should be added only to the DTO that is used for the **request** body